### ISSUE TITLE
[SUGGESTION] Use GRUB/Limine for bootloader

### ISSUE THREAD

**Issue by abxxbo** [cblos/cblos#3]
This might not be the _greatest_ of ideas, but it could work. Instead of rolling our own bootloader, which currently seems a bit problematic, we could use something like [Limine](https://wiki.osdev.org/Limine) or [GRUB](https://wiki.osdev.org/GRUB).

While yes, I _do_ agree that writing our own bootloader would make us more credible as to "we know what we're doing", but my vision is that we get the OS working (shell, GUI [optional]) before we worry about our own bootloader.

**Basically, we would use GRUB until we get everything working, and _then_ write our own bootloader.**

**Comment by xryshirfxn** [comment]
That's a good idea. But I think, it's better to split into 2 teams.

1) Kernel
2) Bootloader

The kernel team will work on the kernel but will use GRUB first. While the bootloader team will work on getting the bootloader to work. Once the bootloader is finished or at least, usable, the kernel team will work on the kernel but using the custom bootloader.

**Comment by abxxbo** [comment]
Since (looking at our devs) we only have 7 people, we might need to have someone be on both kernel & bootloader teams, so it'd be equal.

3 -- just bootloader
3 -- just kernel
1 -- kernel + bootloader

**Comment by xryshirfxn** [comment]
There are some working on the website (2 people, not sure)

**Comment by abxxbo** [comment]
That would still make it unequal (5 people), so in that case,

2 -- bootloader
2 -- kernel
1 -- bootloader + kernel

**Comment by Rstar284** [comment]
we have another person now, so it could be,

2 -- bootloader
2 -- kernel
2 -- bootloader + kernel

**Comment by Rstar284** [comment]
We were gonna use Rust/C for the OS, *i think*.
There exists are crate `bootloader` for rust, we could perhaps use that

**Comment by xryshirfxn** [comment]
Writing one from source is much more fun.

**Comment by Rstar284** [comment]
True.

**Comment by xryshirfxn** [comment]
#4

**Comment by abxxbo** [comment]
Closed because of merging of #4
